### PR TITLE
Rust: Busway Scheme Refactor

### PR DESCRIPTION
**Pull request by droogmic**
In an attempt to make @BudgieInWA 's PR smaller.

### CODE REVIEW

File: rust/osm2lanes/src/transform/tags_to_lanes/modes/bus.rs [hunk 1]
```diff
@@ -59,85 +59,179 @@ pub(in crate::transform::tags_to_lanes) fn bus(
...
-            "opposite_lane",
-        )
-        .into());
+
+    impl Scheme {
+        #[allow(clippy::unnecessary_wraps)]
+        pub(in crate::transform::tags_to_lanes) fn from_tags(
+            tags: &Tags,
+            locale: &Locale,
+            oneway: Oneway,
```
Comment (dabreegster) [inline rust/osm2lanes/src/transform/tags_to_lanes/modes/bus.rs:113]: Takes a oneway and then calculates it below?

**Review by dabreegster — COMMENTED** [review]
Seems like a step in the right direction and preserving the spirit of the other PR, but big warning, I'm rapidly losing the ability to keep up with what's happening here, so my feedback is not great. :\

File: rust/osm2lanes/src/transform/tags_to_lanes/modes/bus.rs (hunk 1 above)
Comment (droogmic) [reply rust/osm2lanes/src/transform/tags_to_lanes/modes/bus.rs:113]: the oneway coming in is for all vehicles, so I need to first check if there is a value specific for busses.

Now that you mention it, the bus-specific preprocessing should maybe happen one abstraction layer further out, but we can do that later.

**Comment by droogmic** [comment]
> Seems like a step in the right direction and preserving the spirit of the other PR, but big warning, I'm rapidly losing the ability to keep up with what's happening here, so my feedback is not great. :\

Yeah, I am just looking for ways to help @BudgieInWA close out his work.

I really want to focus on fixing the widths of small residential roads, but that is mostly dependent on fixing the lane calculation part.

File: rust/osm2lanes/src/transform/tags_to_lanes/modes/bus.rs (hunk 1 above)
Comment (dabreegster) [reply rust/osm2lanes/src/transform/tags_to_lanes/modes/bus.rs:113]: Ah, I didn't read carefully enough. What if we name these `road_oneway` and `bus_oneway`?

**Comment by BudgieInWA** [comment]
> Yeah, I am just looking for ways to help @BudgieInWA close out his work.
> 
> I really want to focus on fixing the widths of small residential roads, but that is mostly dependent on fixing the lane calculation part.

Thank you, you've done a great job of taking each idea and putting it into context, so that we tackle them independently.

**Pull request merged**